cinder/volume/drivers/lightos/lightos.py

```diff
@@ -619,18 +619,22 @@ def _wait_for_volume_available(
         # we can stop on any terminal status
         # possible states: Unknown, Creating, Available, Deleting, Deleted,
         # Failed, Updating
-        states_to_wait_for = (
-            'Available',
-            'Deleting',
-            'Deleted',
-            'Failed',
-            'UNKNOWN')
-        return self._wait_for_volume_state(
-            project_name,
-            timeout=timeout,
-            states=states_to_wait_for,
-            vol_uuid=vol_uuid,
-            vol_name=vol_name)
+        states = ('Available', 'Deleting', 'Deleted', 'Failed', 'UNKNOWN')
+
+        stop = time.time() + timeout
+        while time.time() <= stop:
+            (status_code,
+             resp) = self._get_lightos_volume(project_name,
+                                              timeout=self.logical_op_timeout,
+                                              vol_uuid=vol_uuid,
+                                              vol_name=vol_name)
+            state = resp.get('state', 'UNKNOWN') if \
+                status_code == httpstatus.OK and resp else 'UNKNOWN'
+            if state in states and status_code != httpstatus.NOT_FOUND:
```

Comment:
note that if we fail on PermissionDenied or get invalidarg err not much point in retrying
(we are not supposed to be getting them in happy flow)